packages/flutter/test/widgets/view_test.dart

```diff
@@ -561,6 +561,47 @@ void main() {
     expect(focusNode.hasPrimaryFocus, isTrue);
     expect(FocusManager.instance.rootScope.hasPrimaryFocus, isFalse);
   });
+
+  testWidgets('View notifies engine that a view should have focus when a widget focus change occurs.', (WidgetTester tester) async {
+    final FocusNode nodeA = FocusNode(debugLabel: 'a');
+    addTearDown(nodeA.dispose);
+    final FocusNode nodeB = FocusNode(debugLabel: 'b');
+    addTearDown(nodeB.dispose);
+
+    FlutterView? view;
```

Comment:
Can we add another test: 
- with two views? 
- to see what happens when the focus goes out of all views? 
- to see what happens when the focus goes from the root scope back to a given view?

Reply:
Will do.

Reply:
Done.